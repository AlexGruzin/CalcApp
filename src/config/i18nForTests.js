import i18n from 'i18next';

// locales
import US from '../locales/en';

i18n.init({
  debug: false,
  fallbackLng: ['US'],
  lng: 'US',

  ns: ['auth', 'resetPassword'],
  defaultNS: 'auth',

  resources: {
    US,
  },
});

export default i18n;

