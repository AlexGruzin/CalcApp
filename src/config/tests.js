import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next'
import i18n from "./i18nForTests";
import jsdom from 'jsdom';
const { JSDOM } = jsdom;

/*
 * jest-dom
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom/extend-expect';

/*
 * jest-fetch-mock
 * https://github.com/jefflau/jest-fetch-mock
 */
require('jest-fetch-mock').enableMocks();

if (global.document) {
  global.document.createRange = () => ({
    setStart: () => {
    },
    setEnd: () => {
    },
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
    createContextualFragment: str => JSDOM.fragment(str),
    // createContextualFragment: str => document.createElement('hr'),
  });
  global.open = () => undefined;
}

export const renderWithi18next = Component => {
  const Comp = React.cloneElement(Component, {
    changeLanguage: lng => {
      i18n.changeLanguage(lng);
      rerender(<I18nextProvider i18n={i18n}>{Comp}</I18nextProvider>);
    }
  });
  const defaultRender = render(
    <I18nextProvider i18n={i18n}>{Component}</I18nextProvider>
  );
  const { rerender } = defaultRender;
  return defaultRender;
};


// /**
//  * Wraps a component with an Intl Provider.
//  * @param {Object} component The component to render
//  */
// const withTestIntl = component => <IntlProvider locale="en">{component}</IntlProvider>;
//
// /*
//  * https://testing-library.com/docs/example-react-intl
//  */
// export const renderWithReactIntl = component => render(withTestIntl(component));

