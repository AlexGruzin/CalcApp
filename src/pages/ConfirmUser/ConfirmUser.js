import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { passwordValidation } from 'helpers/validation/auth';
import { useTranslation } from 'react-i18next';
import Button from 'components/Button';
import CustomInput from 'components/CustomInput';

import './ConfirmUser.styles.less';

const ConfirmUser = ({ confirmUser }) => {
  const { t } = useTranslation();
  const { register, handleSubmit: handleSubmitForm } = useForm({ mode: 'onChange', shouldUnregister: true });
  const [user, setUser] = useState({});
  const email = new URLSearchParams(window.location.search).get('email');

  const handleSubmit = () => {
    confirmUser(user);
  };

  const onChangeField = useCallback(e => {
    const { name, value } = e.target;
    setUser(prevState => ({ ...prevState, [name]: value }));
  }, []);

  return (
    <div className="confirm-user">
      <img className="confirm-user__logo" src="/svg/logo.svg" alt="Calc logo" />
      <div className="confirm-user__content">
        <form onSubmit={handleSubmitForm(handleSubmit)} name="confirm-user-form" className="confirm-user__form">
          <CustomInput value={email} disabled type="email" name="email" className="confirm-user__field" />
          <CustomInput
            onChange={onChangeField}
            type="password"
            name="password"
            ref={register(passwordValidation())}
            className="confirm-user__field"
            placeholder={t('confirm:user.password')}
            required
          />
          <CustomInput
            onChange={onChangeField}
            type="text"
            name="firstName"
            className="confirm-user__field"
            placeholder={t('confirm:user.firstName')}
          />
          <CustomInput
            onChange={onChangeField}
            type="text"
            name="lastName"
            className="confirm-user__field"
            placeholder={t('confirm:user.lastName')}
          />
          <Button className="confirm-user__submit" type="submit" title={t('confirm:user.confirm')} />
        </form>
      </div>
    </div>
  );
};

ConfirmUser.propTypes = {
  confirmUser: PropTypes.func.isRequired,
};

export default ConfirmUser;
