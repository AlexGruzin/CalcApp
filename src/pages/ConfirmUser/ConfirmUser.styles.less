@import "~styles/generic.less";

.confirm-user {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
}

.confirm-user__logo {
  .calc-logo();
}

.confirm-user__content {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin-bottom: 100px;
}

.confirm-user__form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: var(--space-x2);
  width: 390px;
  min-width: 390px;
  max-width: 390px;
  height: 397px;
  max-height: 397px;
  min-height: 397px;
  background-color: var(--white);
  border-radius: 13px;
}

.confirm-user__field input {
  height: 36px;
  width: 294px;
  text-align: center;
  border-radius: 30px;
  border: 1px solid var(--border-grey);
  margin-bottom: 23px;
}

.confirm-user__submit {
  height: 35px;
  min-height: 35px;
  width: 162px;
  border-radius: 18px;
  background-color: var(--button-purple);
  text-transform: capitalize;
  margin-bottom: var(--space-x3);
}
