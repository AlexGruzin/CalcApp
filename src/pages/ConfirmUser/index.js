import { connect } from 'react-redux';

import { initConfirmUser } from 'sagas/actions/confirmUser';
import ConfirmUser from './ConfirmUser';

export default connect(null, { confirmUser: initConfirmUser })(ConfirmUser);
