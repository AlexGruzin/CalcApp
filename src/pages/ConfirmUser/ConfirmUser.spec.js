import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithi18next } from 'config/tests';
import ConfirmUser from './ConfirmUser';

describe('ConfirmUser tests', () => {
  it('Check snapshots of ConfirmUser are equal.', async () => {
    const { asFragment } = render(<ConfirmUser confirmUser={() => undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Press submit to send api request (new member) with valid password.', async () => {
    const mockSubmit = jest.fn();
    const { getAllByRole, getByLabelText, asFragment } = renderWithi18next(<ConfirmUser confirmUser={mockSubmit} />);
    const btns = getAllByRole('button');
    const input = getByLabelText('password');

    fireEvent.change(input, { target: { value: '12345678' } }); // enter password
    userEvent.click(btns[0]);

    waitFor(() => {
      expect(mockSubmit).toBeCalledTimes(1);
    });
  });

  it('Press submit to send api request (new member) with invalid password.', async () => {
    const mockSubmit = jest.fn();
    const { getAllByRole, getByLabelText } = renderWithi18next(<ConfirmUser confirmUser={mockSubmit} />);
    const btns = getAllByRole('button');
    const input = getByLabelText('password');

    fireEvent.change(input, { target: { value: '1' } }); // enter password

    userEvent.click(btns[0]);

    waitFor(() => {
      expect(mockSubmit).toBeCalledTimes(0);
    });
  });
});
