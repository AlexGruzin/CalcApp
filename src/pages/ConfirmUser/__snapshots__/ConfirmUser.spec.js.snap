// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmUser tests Check snapshots of ConfirmUser are equal. 1`] = `
<DocumentFragment>
  <div
    class="confirm-user"
  >
    <img
      alt="Calc logo"
      class="confirm-user__logo"
      src="/svg/logo.svg"
    />
    <div
      class="confirm-user__content"
    >
      <form
        class="confirm-user__form"
        name="confirm-user-form"
      >
        <div
          class="confirm-user__field custom-input"
        >
          <input
            aria-label="email"
            disabled=""
            id="email"
            name="email"
            type="email"
            value=""
          />
          <div
            class="custom-input__error"
          />
        </div>
        <div
          class="confirm-user__field custom-input"
        >
          <input
            aria-label="password"
            id="password"
            name="password"
            placeholder="confirm:user.password"
            required=""
            type="password"
          />
          <div
            class="custom-input__error"
          />
        </div>
        <div
          class="confirm-user__field custom-input"
        >
          <input
            aria-label="firstName"
            id="firstName"
            name="firstName"
            placeholder="confirm:user.firstName"
            type="text"
          />
          <div
            class="custom-input__error"
          />
        </div>
        <div
          class="confirm-user__field custom-input"
        >
          <input
            aria-label="lastName"
            id="lastName"
            name="lastName"
            placeholder="confirm:user.lastName"
            type="text"
          />
          <div
            class="custom-input__error"
          />
        </div>
        <button
          class="custom-button confirm-user__submit"
          type="submit"
          value=""
        >
          confirm:user.confirm
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
